src/features/Folding.ts

```diff
@@ -0,0 +1,419 @@
+import * as path from "path";
+import * as vscode from "vscode";
+import {
+    DocumentSelector,
+    LanguageClient,
+} from "vscode-languageclient";
+import { IFeature } from "../feature";
+import * as Settings from "../settings";
+
+const tm = getCoreNodeModule("vscode-textmate");
+// Returns a node module installed with VSCode, or null if it fails.
+// https://github.com/Microsoft/vscode/issues/46281
+function getCoreNodeModule(moduleName: string) {
+    try {
+        return require(`${vscode.env.appRoot}/node_modules.asar/${moduleName}`);
+    // tslint:disable-next-line:no-empty
+    } catch (err) { }
+
+    try {
+        return require(`${vscode.env.appRoot}/node_modules/${moduleName}`);
+    // tslint:disable-next-line:no-empty
+    } catch (err) { }
+
+    return null;
+}
+
+interface IExtensionGrammar {
+    language?: string;
+    scopeName?: string;
+    path?: string;
+    extensionPath?: string;
+    embeddedLanguages?: {
+        [scopeName: string]: string,
+    }; injectTo?: string[];
```

Comment:
Should the `injectTo?` field be pushed down onto its own line?

Reply:
Fixed.